flume-ng-sinks/flume-dataset-sink/src/main/java/org/apache/flume/sink/kite/DatasetSink.java

```diff
@@ -48,144 +39,157 @@
-    String effectiveUser =
-        context.getString(DatasetSinkConstants.AUTH_PROXY_USER);
+        context.getString(AUTH_PRINCIPAL),
+        context.getString(AUTH_KEYTAB));
+    String effectiveUser
+        = context.getString(AUTH_PROXY_USER);
     if (effectiveUser != null) {
       this.login = KerberosUtil.proxyAs(effectiveUser, login);
     }
 
-    String datasetURI = context.getString(
-      DatasetSinkConstants.CONFIG_KITE_DATASET_URI);
+    // Get the dataset URI and name from the context
+    String datasetURI = context.getString(CONFIG_KITE_DATASET_URI);
     if (datasetURI != null) {
-      this.target = URI.create(datasetURI);
-      this.datasetName = uriToName(target);
+      this.datasetUri = URI.create(datasetURI);
+      this.datasetName = uriToName(datasetUri);
     } else {
-      String repositoryURI = context.getString(
-        DatasetSinkConstants.CONFIG_KITE_REPO_URI);
-      Preconditions.checkNotNull(repositoryURI, "Repository URI is missing");
-      this.datasetName = context.getString(
-        DatasetSinkConstants.CONFIG_KITE_DATASET_NAME);
-      Preconditions.checkNotNull(datasetName, "Dataset name is missing");
-
-      this.target = new URIBuilder(repositoryURI, datasetName).build();
+      String repositoryURI = context.getString(CONFIG_KITE_REPO_URI);
+      Preconditions.checkNotNull(repositoryURI, "Repository URI is missing."
+          + " You must set either " + CONFIG_KITE_DATASET_URI + " or both"
+          + " " + CONFIG_KITE_REPO_URI + " and " + CONFIG_KITE_DATASET_NAME);
+
+      this.datasetName = context.getString(CONFIG_KITE_DATASET_NAME);
+      Preconditions.checkNotNull(datasetName, "Dataset name is missing."
+          + " You must set either " + CONFIG_KITE_DATASET_URI + " or both"
+          + " " + CONFIG_KITE_REPO_URI + " and " + CONFIG_KITE_DATASET_NAME);
+
+      String namespace = context.getString(CONFIG_KITE_DATASET_NAMESPACE,
+          DEFAULT_NAMESPACE);
+
+      this.datasetUri = new URIBuilder(repositoryURI, namespace, datasetName)
+          .build();
     }
+    this.setName(datasetUri.toString());
 
-    this.setName(target.toString());
+    // Create the configured failure policy
+    this.policy = policyFactory.newPolicy(context);
 
     // other configuration
-    this.batchSize = context.getLong(
-        DatasetSinkConstants.CONFIG_KITE_BATCH_SIZE,
-        DatasetSinkConstants.DEFAULT_BATCH_SIZE);
-    this.rollIntervalS = context.getInteger(
-        DatasetSinkConstants.CONFIG_KITE_ROLL_INTERVAL,
-        DatasetSinkConstants.DEFAULT_ROLL_INTERVAL);
+    this.batchSize = context.getLong(CONFIG_KITE_BATCH_SIZE,
+        DEFAULT_BATCH_SIZE);
+    this.rollIntervalS = context.getInteger(CONFIG_KITE_ROLL_INTERVAL,
+        DEFAULT_ROLL_INTERVAL);
 
     this.counter = new SinkCounter(datasetName);
   }
 
   @Override
   public synchronized void start() {
-    this.lastRolledMs = System.currentTimeMillis();
+    lastRolledMs = System.currentTimeMillis();
```

Comment:
Why did you remove the reference to `this`? I consider it best practice to call out when you are setting an instance variable.